### PR TITLE
Add concurrency test for PendingTrace

### PR DESCRIPTION
**Pull request by tylerbenson**
Before the fix in #2086, this test fails reliably if I add a sleep between these two lines in `PendingTrace`:
```
    finishedSpans.addFirst(span);
    completedSpanCount.incrementAndGet();
```

### CODE REVIEW

File: dd-java-agent/agent-logging/src/test/groovy/datadog/trace/logging/GlobalLogLevelSwitcherTest.groovy [hunk 1]
```diff
@@ -8,7 +8,7 @@ import datadog.trace.logging.simplelogger.SLCompatSettings
 import org.slf4j.ILoggerFactory
 import org.slf4j.Logger
 
-class GlobalLogeLevelSwitcherTest extends LogValidatingSpecification {
+class GlobalLogLevelSwitcherTest extends LogValidatingSpecification {
```
Comment (tylerbenson) [inline dd-java-agent/agent-logging/src/test/groovy/datadog/trace/logging/GlobalLogLevelSwitcherTest.groovy:11]: This change is unrelated... just fixing a simple typo and formatting.

**Review by richardstartin — APPROVED** [review]

**Review by bantonsson — APPROVED** [review]
